backend/routes/addUnknowns.js

```diff
@@ -0,0 +1,96 @@
+const express = require('express');
+const router = express.Router();
+const admin = require('firebase-admin');
+
+// Add Unknowns Endpoint
+router.post('/addUnknowns', async (req, res) => {
+  try {
+    // Extract location and occupied courts from the request body
+    const { location, occupiedCourts } = req.body;
+    if (!location || !Array.isArray(occupiedCourts)) {
+      return res.status(400).json({ message: 'Location and occupied court information is required.' });
+    }
+
+    // Retrieve active court information for the given location
+    const activePlayersSnapshot = await admin.firestore()
+      .collection('locations')
+      .doc(location)
+      .collection('activePlayers')
+      .orderBy('courtNumber')
+      .get();
+
+    // Check for empty active player snapshot (as always filled with Empty/Unknown/Player objects)
+    if (activePlayersSnapshot.empty) {
+      return res.status(404).json({ message: 'Invalid location.' });
+    }
+
+    // Creating shuffled timestamp adjustments
+    const now = new Date();
+    const adjustments = [-40, -30, -20, -10, 0];
+    const timestamps = adjustments.map(seconds => {
+      const adjustedTime = new Date(now);
+      adjustedTime.setSeconds(now.getSeconds() + seconds);
+      return adjustedTime;
+    });
+    shuffleArray(timestamps);
+    let lastTimestampIdx = 0;
+
+    // Update each document based on user information
+    const batch = admin.firestore().batch();
+    activePlayersSnapshot.docs.forEach(doc => {
```

Comment:
I saw in this [documentation](https://firebase.google.com/docs/reference/node/firebase.firestore.QuerySnapshot#foreach) that there is a forEach method for the snapshot. I think there's no need to have `.docs`.